bson/src/main/org/bson/io/ByteBufferBsonInput.java

```diff
@@ -182,11 +183,64 @@ public void skipCString() {
         buffer.position(pos + length);
     }
 
+    /**
+     * Detects the position of the first NULL (0x00) byte in a 64-bit word using SWAR technique.
+     * <a href="https://en.wikipedia.org/wiki/SWAR">
+     */
     private int computeCStringLength(final int prevPos) {
-        ensureOpen();
-        int pos = buffer.position();
+        int pos = prevPos;
         int limit = buffer.limit();
 
+        // `>>> 3` means dividing without remainder by `Long.BYTES` because `Long.BYTES` is 2^3
+        int chunks = (limit - pos) >>> 3;
+        // `<< 3` means multiplying by `Long.BYTES` because `Long.BYTES` is 2^3
+        int toPos = pos + (chunks << 3);
+        for (; pos < toPos; pos += Long.BYTES) {
+            long chunk = buffer.getLong(pos);
+            /*
+              Subtract 0x0101010101010101L to cause a borrow on 0x00 bytes.
+              if original byte is 00000000, then 00000000 - 00000001 = 11111111 (borrow causes the most significant bit set to 1).
+             */
+            long mask = chunk - 0x0101010101010101L;
+            /*
+              mask will only have the most significant bit set iff it was a 0x00 byte (0x00 becomes 0xFF because of the borrow).
```

Comment:
"most significant bit" may refer to the most significant bit in a byte, or in a long (the chunk), and without clarified the text is still unclear. Note also how you used "significant byte **of the long**" below.

```suggestion
              mask will only have the most significant bit in each byte set iff it was a 0x00 byte (0x00 becomes 0xFF because of the borrow).
```